core/src/main/scala/org/apache/spark/rdd/InputFileBlockHolder.scala

```diff
@@ -41,25 +43,26 @@ private[spark] object InputFileBlockHolder {
    * The thread variable for the name of the current file being read. This is used by
    * the InputFileName function in Spark SQL.
```

Comment:
It may not intuitive to understand this thread local variable now, can we add a short comment to explain why an `AtomicReference` here?